plugins/chat/app/controllers/chat/api/channel_messages_controller.rb

```diff
@@ -4,4 +4,10 @@ class Chat::Api::ChannelMessagesController < Chat::ApiController
   def destroy
     with_service(Chat::TrashMessage) { on_model_not_found(:message) { raise Discourse::NotFound } }
   end
+
+  def restore
+    with_service(Chat::RestoreMessage) do
+      on_model_not_found(:message) { raise Discourse::NotFound }
+    end
+  end
```

Comment:
Technically I would prefer `POST /channel/:channel_id/messages/:message_id/restores`, but up to you this is easy to change in the future if you don't want to bother for now

Reply:
Yeah not sure I want to bother with this for now. REST is so weird since we're not creating anything with restore, but I get your meaning.